packages/react-i18n/README.md

```diff
@@ -232,6 +225,32 @@ i18n.translate('MyComponent.searchResult', {
 });
 ```
 
+#### `withI18n` decorator
```

Comment:
I feel like we need a slightly better story here. Right now, this is great for legacy purposes, which you've noted in the doc. However, this would also be the only option for class-based components. How would you feel about adding a `<WithI18n>` component that provides the i18n instance in a render prop (this brings it in to a class-based component in a similar way to how it will end up looking in a function component, though obviously with more nesting)? Then the docs could be structured basically like `The right way` => `If you need classes` => `If you have legacy code`